lib/point_quest/quests/commands/sort_objective.ex

```diff
@@ -0,0 +1,51 @@
+defmodule PointQuest.Quests.Commands.SortObjective do
+  @moduledoc """
+  Updates the sort order for a single objective in a quest.
+  """
+  use PointQuest.Valuable
+
+  alias PointQuest.Quests
+
+  require PointQuest.Quests.Telemetry
+  require Telemetrex
+
+  @type t :: %__MODULE__{
+          quest_id: String.t(),
+          objective_id: String.t(),
+          sort_value: float()
+        }
+
+  @primary_key false
+  embedded_schema do
+    field :quest_id, :string
+    field :objective_id, :string
+    field :sort_value, :float
+  end
+
+  @spec execute(t(), PointQuest.Authentication.Actor.t()) ::
+          {:ok, PointQuest.Quests.Event.ObjectiveSorted.t()}
+  def execute(%__MODULE__{quest_id: quest_id} = sort_objective_command, actor) do
+    Telemetrex.span event: Quests.Telemetry.objective_sorted(),
+                    context: %{command: sort_objective_command, actor: actor} do
+      with {:ok, quest} <- PointQuest.quest_repo().get_quest_by_id(quest_id),
+           true <- can_sort?(quest, actor),
+           {:ok, event} <- Quests.Quest.handle(sort_objective_command, quest),
+           :ok <- PointQuest.quest_repo().write(quest, event) do
+        {:ok, event}
+      else
+        false -> {:error, :must_be_leader_of_party}
```

Comment:
Idea for later. Just like our generic not found error we should make a not authorized error we can return in these cases.